catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/report/RegistryReportBuilderTest.java

```diff
@@ -0,0 +1,455 @@
+                .item(23)
+                .getTextContent();
+        String serviceURLValue = htmlDoc.getElementsByTagName("td")
+                .item(47)
+                .getTextContent();
+
+        assertThat(serviceName, is("Service Name"));
+        assertThat(descriptionName, is("Description"));
+        assertThat(descriptionValue, is("Non Federation Node Extrinsic Object Description"));
+        assertThat(serviceBindingName, is("CSW Federation Method"));
+        assertThat(serviceURLName, is("cswUrl"));
+        assertThat(serviceURLValue, is("https://some/address/here"));
+    }
+
+    private void assertOrganizations(Document htmlDoc) {
+        String organizationName = htmlDoc.getElementsByTagName("h3")
+                .item(1)
+                .getTextContent();
+        String statusName = htmlDoc.getElementsByTagName("h4")
+                .item(1)
+                .getTextContent();
+        String statusValue = htmlDoc.getElementsByTagName("td")
+                .item(50)
+                .getTextContent();
+        String emailName = htmlDoc.getElementsByTagName("h4")
+                .item(13)
+                .getTextContent();
+        String emailValue = htmlDoc.getElementsByTagName("td")
+                .item(62)
+                .getTextContent();
+
+        assertThat(organizationName, is("Codice"));
+        assertThat(statusName, is("Status"));
+        assertThat(statusValue, is("someStatus"));
+        assertThat(emailName, is("Email Addresses"));
+        assertThat(emailValue, is("emailaddress@something.com"));
+    }
+
+    private void assertContacts(Document htmlDoc) {
+        String contactName = htmlDoc.getElementsByTagName("h3")
+                .item(2)
+                .getTextContent();
+        String emailName = htmlDoc.getElementsByTagName("h4")
+                .item(17)
+                .getTextContent();
+        String emailValue = htmlDoc.getElementsByTagName("td")
+                .item(66)
+                .getTextContent();
+
+        assertThat(contactName, is("foo bar"));
+        assertThat(emailName, is("Email Addresses"));
+        assertThat(emailValue, is("emailaddress@something.com"));
+    }
+
+    private void assertCollections(Document htmlDoc) {
+        String collectionName = htmlDoc.getElementsByTagName("h3")
+                .item(3)
+                .getTextContent();
+        String versionInfoName = htmlDoc.getElementsByTagName("strong")
+                .item(25)
+                .getTextContent();
+        String versionInfoValue = htmlDoc.getElementsByTagName("td")
+                .item(68)
+                .getTextContent();
+
+        assertThat(collectionName, is("Another Name"));
+        assertThat(versionInfoName, is("VersionInfo"));
+        assertThat(versionInfoValue, is("2.10.x"));
+    }
+
+    private Document xmlFileToDocument(String html) throws Exception {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
You may want to consider using try with resources in here to account for Closable implementations that need to get cleaned-up.